src/pysdccc/_cli.py

```diff
@@ -0,0 +1,133 @@
+import contextlib
+import io
+import pathlib
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+import httpx
+
+from pysdccc import _download, _runner
+
+try:
+    import click
+except ImportError as import_error:
+    msg = "Cli not installed. Please install using 'pip install pysdccc[cli]."
+    raise ImportError(msg) from import_error
+
+
+class UrlType(click.ParamType):
+    name = 'url'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.URL:
+        try:
+            return httpx.URL(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid url: {e}', param, ctx)
+
+
+URL = UrlType()
+
+
+class ProxyType(click.ParamType):
+    name = 'proxy'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.Proxy:
+        try:
+            return httpx.Proxy(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid proxy: {e}', param, ctx)
+
+
+PROXY = ProxyType()
+
+
+def _download_to_stream(url: httpx.URL, stream: io.IOBase, proxy: httpx.Proxy | None = None) -> None:
+    with _download.open_download_stream(url, proxy) as response:
+        total = response.headers.get('Content-Length')
+        with click.progressbar(
+                response.iter_bytes(),
+            length=int(total) if total is not None else None, label='Downloading', show_eta=True, width=0
+        ) as progress:
+            num_bytes_downloaded = response.num_bytes_downloaded
+            for chunk in progress:
+                stream.write(chunk)
+                progress.update(response.num_bytes_downloaded - num_bytes_downloaded)
+                num_bytes_downloaded = response.num_bytes_downloaded
+
+
+def download(url: httpx.URL, output: pathlib.Path, proxy: httpx.Proxy | None = None):
+    click.echo(f'Downloading sdccc from {url}.')
+    with tempfile.NamedTemporaryFile('wb', suffix='.zip', delete=False) as temporary_file:
+        _download_to_stream(url, temporary_file, proxy=proxy)  # type: ignore[arg-type]
+    click.echo(f'Extracting sdccc to {output}.')
+    with (
+        zipfile.ZipFile(temporary_file.name) as f,
+        click.progressbar(f.infolist(), label='Extracting', width=0) as progress,
+    ):
+        for member in progress:
+            f.extract(member, output)
+
+
+@click.group(help='Manage sdccc installation.')
+@click.version_option(message='%(version)s')
+def cli():
+    pass
+
+
+@click.command(
+    short_help='Install the SDCcc executable from the specified URL. Releases can be found at https://github.com/Draegerwerk/SDCcc/releases.',
```

Comment:
line too long?!

Reply:
if ruff is not complaining, its not too long :D